src/Geometry/Primitives/NonCSG/ConvexPolygon.jl

```diff
@@ -0,0 +1,104 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+import Statistics
+import GeometricalPredicates
+
+"""
+    ConvexPolygon{T} <: Surface{T}
+
+General Convex Polygon surface, not a valid CSG object.
+The rotation of the polygon around its normal is defined by `rotationvec`.
+`rotationvec×surfacenormal` is taken as the vector along the u axis.
+
+```julia
+ConvexPolygon(local_frame::Transform{T}, local_polygon_points::Vector{Vector{T}}, interface::NullOrFresnel{T} = nullinterface(T))
+```
+
+The local frame defines the plane (spans by the right and up vectors) with the plane normal given by the forward vector.
+the local_polygon_points are given with respect to the local frame and are 2D points.
+"""
+struct ConvexPolygon{T} <: Surface{T}
+    plane::Plane{T,3}
+    local_frame::Transform{T}
+    local_points::Vector{Vector{T}}
+
+    # for efficency
+    _poly2d::GeometricalPredicates.Polygon2D{GeometricalPredicates.Point2D}
+
+    function ConvexPolygon(
+            local_frame::Transform{T},
+            local_polygon_points::Vector{Vector{T}}, 
+            interface::NullOrFresnel{T} = NullInterface(T)
+        ) where {T<:Real}
+
+        @assert length(local_polygon_points) > 3         # need at least 3 points to define apolygon
+        @assert size(local_polygon_points[1])[1] == 2    # check that first point is a 2D point
+
+        local_center = Statistics.mean(local_polygon_points)
+        world_center = local2world(local_frame) * Vec3(local_center[1], local_center[2], zero(T))
+
+        poly_points = [GeometricalPredicates.Point2D(p[1], p[2]) for p in local_polygon_points]
+        poly = GeometricalPredicates.Polygon2D(poly_points...)
+
+        plane = Plane(forward(local_frame), world_center, interface = interface)
+        new{T}(plane, local_frame, local_polygon_points, poly)
+    end
+end
+export ConvexPolygon
+
+# Base.show(io::IO, poly::ConvexPolygon{T}) where {T<:Real} = print(io, "ConvexPolygon{$T}($(centroid(hex)), $(normal(hex)), $(hex.side_length), $(interface(hex)))")
+centroid(poly::ConvexPolygon{T}) where {T<:Real} = poly.plane.pointonplane
+interface(poly::ConvexPolygon{T}) where {T<:Real} = interface(poly.plane)
+normal(poly::ConvexPolygon{T}) where {T<:Real} = normal(poly.plane)
+normal(poly::ConvexPolygon{T}, ::T, ::T) where {T<:Real} = normal(poly)
```

Comment:
what is this function for? If you really need two then the reason should probably be in a documentation string for the normal function.

Reply:
I copied it from the Hexagon example. I think that its just a matter of interface support.
You can have a normal for the surface and a normal for each point on the surface in the case when it's not planer.

Reply:
It seems that all the shapes have the same interface - i think that normal without parameters is the one that should be removed but on the other hand what point would you give to a shape that doesn't have a natural parametrization.